providers/curation/github.js

```diff
@@ -109,13 +109,38 @@ class GitHubCurationService {
     // Github requires name/email to set committer
     if ((info.name || info.login) && info.email)
       fileBody.committer = { name: info.name || info.login, email: info.email }
+    else {
+      logger.info('Github requires name/email to set committer, info.email = ', info.email)
+    }
     if (get(currentContent, '_origin.sha')) {
       fileBody.sha = currentContent._origin.sha
+      // Extract the component values
+      const { coordinates, revisions } = currentContent
+      const { name, provider, type } = coordinates
+      let keys = Object.keys(revisions)
+      keys.forEach(revision => {
+        let component = name + '/' + revision
+        if(!this._validateComponentExists(component)){
```

Comment:
Given this is an `async` function, perhaps you should wait for the result before check it.
So this could become `const validateResult = await this._validateComponentExists(component)` or 
`this._validateComponentExists(component).then(validateResult => {})`